src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class Storage {
+    private String taskListFileName;
+    private File taskListFile;
+
+    /**
+     * Handles saving of taskLists between Duke runtime sessions to text file by serialisation.
+     * @param taskListFileName path to text file to store taskList in.
+     */
+    public Storage(String taskListFileName) {
+        this.taskListFileName = taskListFileName;
+        try {
+            Files.createDirectories(Paths.get(this.taskListFileName).getParent().getFileName());
+        } catch (IOException e) {
+            System.out.println("cannot create directory");
+        }
+        this.taskListFile = new File(this.taskListFileName);
+    }
+
+    /**
+     * Loads taskList from textFile configured in constructor if it exists, else initialises empty taskList.
+     * @return stored taskList if present, or empty taskList.
+     */
+    public TaskList initialise() {
+        TaskList taskList = new TaskList();
+        try {
+            if (!this.taskListFile.createNewFile()) {
+                FileInputStream fis = new FileInputStream(this.taskListFileName);
+                ObjectInputStream ois = new ObjectInputStream(fis);
```

Comment:
Perhaps it would be better if more descriptive/intuitive variable names were used here? I noticed this in the `store` method as well.

Reply:
Since the names are only used within this logical block, following the principle of short lifespan = short name I believe that these are an appropriate names